go/prometheusexporter/collector/collector.go

```diff
@@ -741,14 +741,20 @@ func (c *Collector) recordLogMetrics(m *metric) {
 	switch metric {
 	case "http":
 		p["type"] = c.requestType(action)
+		q := removeAction(p)
 		c.actionMetrics.httpRequestSummaryVec.With(p).Observe(m.value)
+		c.applicationMetrics.httpRequestSummaryVec.With(q).Observe(m.value)
```

Comment:
Could we extract this block of code and call it for `c.actionMetrics` and `c.applicationMetrics` ?

Reply:
Possibly. Let me give it a whirl.